### PR TITLE
Remove most checkstyle violations in test code

### PR DESCRIPTION
**Pull request by RoiEXLab**
There are only few violations left in the test code.
We don't need to worry about the SHA512 violation, #2101 removes that class.
There is some javadoc needed in the GameDataTestUtil class, I tried adding javadocs as best as I could, but I wasn't able to figure out what some methods do.
Also there are some violations with camelCase package names.
I tried changing those, but this causes errors on case insensitive file systems. (I wasn't able to make eclipse compile triplea)

### CODE REVIEW

File: src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java [hunk 1]
```diff
@@ -292,7 +292,7 @@ private static String validateAttachment(final Class<? extends IAttachment> claz
       // validate that there is a field and a getter
       Field field = null;
       try {
-        field = PropertyUtil.getFieldIncludingFromSuperClasses(clazz, "m_" + propertyName, false);
+        field = PropertyUtil.getFieldIncludingFromSuperClasses(clazz, propertyName, false);
```
Comment (ssoloff) [inline src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java]: I looked at this once before when I first started working on TripleA.  I didn't modify it at the time because I wasn't sure where the requirement for the `m_` prefix came from.  I thought there must be some connection to the production code, but I couldn't find any at the time.

If this test class is not being run against any of our production attachments, then does anyone know what its purpose actually is?  It doesn't seem useful to test that a bunch of test-only annotations obey a test-only validation class.  Am I missing something?

File: src/test/java/games/strategy/engine/data/annotations/InvalidFieldNameExample.java [hunk 2]
```diff
@@ -16,15 +16,15 @@ protected InvalidFieldNameExample(final String name, final Attachable attachable
   }
 
   // should have been prefixed with "m_". Should cause test to fail.
```
Comment (ssoloff) [inline src/test/java/games/strategy/engine/data/annotations/InvalidFieldNameExample.java]: This comment should probably be updated to reflect that it's the fact that the field has a different name which is causing the test to fail rather than the lack of the `m_` prefix.

File: src/test/java/games/strategy/triplea/delegate/GameDataTestUtil.java [hunk 3]
```diff
@@ -245,10 +361,17 @@ public static void givePlayerRadar(final PlayerID player) {
...
   public static void assertValid(final String string) {
     Assert.assertNull(string, string);
   }
-
+  
+  /**
+   * Helper method to check if a String is null and otherwise print the String
+   * which doesn't make any sense because the String will then always be null.
+   */
   public static void assertError(final String string) {
```
Comment (ssoloff) [inline src/test/java/games/strategy/triplea/delegate/GameDataTestUtil.java]: I think this method is asserting that an error is present, where the specified string is probably an error message or something.  That is, if no error occurred, the string would be `null`; if an error occurred, the string would be non-`null`.

Regardless, if you keep the existing Javadoc, the phrase "if a String is null" in the first sentence should probably be "if a String is **not** null".

File: src/test/java/games/strategy/triplea/delegate/WW2V341Test.java [hunk 4]
```diff
@@ -95,7 +95,7 @@
 import games.strategy.util.IntegerMap;
 import games.strategy.util.Match;
 
-public class WW2V3_41_Test {
+public class WW2V341Test {
```
Comment (ssoloff) [inline src/test/java/games/strategy/triplea/delegate/WW2V341Test.java]: I'm puzzled as to why this isn't triggering a Checkstyle abbreviation violation.  Oh well. :smile:

The "V341" kind of runs together here.  I don't think there's any danger of someone not being able to parse this mentally, but naming these types of tests something like `WorldWar2Version3Year1941Test` might be easier on the eyes (if just a bit verbose).

File: src/test/java/games/strategy/util/PropertyUtilTest.java [hunk 5]
```diff
@@ -54,7 +54,7 @@ public void testGetField() {
 
   @Test
   public void testGetFieldWithPrefixedPropertyNames() {
-    assertThat(PropertyUtil.getPropertyFieldObject(BAR, new mUnderBarClass()), is(DEFAULT));
+    assertThat(PropertyUtil.getPropertyFieldObject(BAR, new UnderBarClass()), is(DEFAULT));
```
Comment (ssoloff) [inline src/test/java/games/strategy/util/PropertyUtilTest.java]: This test looks like it's specifically verifying that the `PropertyUtil#getPropertyFieldObject()` method can find a field with an `m_` prefix.  I don't think the `UnderBarClass#m_bar` field should be renamed.  We can always suppress this warning when we introduce Checkstyle suppression filters.

However, it seems perfectly fine to have renamed `mUnderBarClass` to `UnderBarClass`.  `PropertyUtil` doesn't seem to have any dependency on the class name.

File: src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java (hunk 1 above)
Comment (ssoloff) [inline src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java]: The `fail()` call was commented out in 12981f3 (circa 2012).  So this validation has been ignored for quite some time.

As a result of turning off the test failures, the `testAllAttachments()` test is broken, as it currently doesn't find any attachment classes to validate because the compilation output folder changed at some time in the past (probably during the conversion to Gradle).

Comment (ssoloff) [inline src/test/java/games/strategy/engine/data/annotations/ValidateAttachmentsTest.java]: It looks like the `PropertyUtil` class handles searching for fields with or without the `m_` prefix via the `PropertyUtil#getPropertyFieldObject()`.  However, the `PropertyUtil#getFieldIncludingFromSuperClasses()` method, used in the above test code, only looks for a field with the given name without trying the additional `m_` prefix.  Unfortunately, it's not straightforward to swap one method call for the other.  In fact, it looks like it would require some significant refactoring to get this to work as expected.

With that said, I'm reconsidering my recommendation that the changes to the test attachment classes be reverted.  If `PropertyUtil` can handle either prefixed or unprefixed fields, then the test simply needs to be updated.  Granted, this change reports about 400 more warnings to the console than previously because a lot of the production attachment classes use the `m_` prefix for fields.  But it seems like the right thing to do is fix the test in a separate PR instead of reverting the changes in this package.

**TL;DR** The changes in the `src/test/java/games/strategy/engine/data/annotations` folder are fine.  The only thing that needs to be addressed is updating the comment in `InvalidFieldNameExample`.

**Comment by RoiEXLab** [comment]
@ssoloff I updated the code to adress your issues
